log4j-api-test/src/test/java/org/apache/logging/log4j/status/StatusLoggerPropertiesUtilDoubleTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.status;
+
+import static org.apache.logging.log4j.status.StatusLogger.PropertiesUtilsDouble.readAllAvailableProperties;
+import static org.apache.logging.log4j.status.StatusLogger.PropertiesUtilsDouble.readProperty;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.org.webcompere.systemstubs.SystemStubs.restoreSystemProperties;
+import static uk.org.webcompere.systemstubs.SystemStubs.withEnvironmentVariable;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.parallel.ResourceAccessMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+import org.junit.jupiter.api.parallel.Resources;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class StatusLoggerPropertiesUtilDoubleTest {
+
+    private static final String[] NOT_PREFIXED_MATCHING_PROPERTY_NAMES = new String[] {
+        // For system properties:
+        "StatusLogger.DateFormat",
+        "StatusLogger.dateFormat",
+        "StatusLogger.dateformat",
+        "StatusLogger.Dateformat",
+        "Status.Logger.Dateformat",
+        "Status.Logger.Date.Format",
+        "statusLoggerDateFormat",
+        "Status.Logger.Date.Format",
+        "status.logger.date.format",
+        // For environment variables:
+        "STATUSLOGGER_DATEFORMAT",
+        "STATUS_LOGGER_DATE_FORMAT"
+    };
```

Comment:
I would relax the test to only check these cases:

- the legacy (pre-2.10) values `log4j2.StatusLogger.level` and `log4j2.StatusLogger.DateFormat`,
- the 2.10 system properties values `log4j2.statusLoggerLevel` and `log4j2.statusLoggerDateFormat`,
- the 2.10 environment variables values `LOG4J_STATUS_LOGGER_LEVEL` and `LOG4J_STATUS_LOGGER_DATE_FORMAT`,
- future (3.0) values `log4j.StatusLogger.level` and `log4j.StatusLogger.dateFormat`.

I believe we should test the externally mandated behavior, in such a way that the test does not require any changes if the internal implementation changes.

Since we don't officially support `log4j2.status.logger.date.format`, we don't accept issue reports if it stops working and we should probably not test for it.

(Reminder to myself: I overspecified a lot of tests too)

Reply:
Fixed in 9c52bfebd8e2292c1eb549f058ae16394de009f8.